lms/djangoapps/discussion/rest_api/serializers.py

```diff
@@ -108,6 +109,19 @@ def __init__(self, *args, **kwargs):
         for field in self.non_updatable_fields:
             setattr(self, "validate_{}".format(field), self._validate_non_updatable)
 
+    def to_representation(self, obj):
+        # get the original representation
+        ret = super(_ContentSerializer, self).to_representation(obj)
+
+        # remove staff only fields if not staff
+        if self.context.get('is_staff', False) is not True:
+            for staff_only_field in self.staff_only_fields:
+                try:
+                    ret.pop(staff_only_field)
+                except KeyError:
+                    continue
+        return ret
```

Comment:
Is this really needed?
You can let this field exist, but just set it to None if accessed as non-staff.